docs/machine-learning/tutorials/tensorflow-transfer-learning.md

```diff
@@ -0,0 +1,337 @@
+{
+
+}
+```
+
+## Extract Features and transform the data
+
+Pre-processing and cleaning data are important tasks that occur before a dataset is used effectively for machine learning. Raw data is often noisy and unreliable, and may be missing values. Using data without these modeling tasks can produce misleading results.
+
+ML.NET's transform pipelines compose a custom `transforms`set that is applied to your data before training or testing. The transforms' primary purpose is data [featurization](../resources/glossary.md#feature-engineering). Machine learning algorithms understand [featurized](../resources/glossary.md#feature) data, so the next step is to transform our textual data into a format that our ML algorithms recognize. That format is a [numeric vector](../resources/glossary.md#numerical-feature-vector).
+
+When the model is trained and evaluated, by default, the values in the **Label** column are considered as correct values to be predicted. As you are using a pretrained model you need to map fields to the new model.. To do that, use the `MLContext.Transforms.Conversion.MapValueToKey`, which is a wrapper for the <xref:Microsoft.ML.ConversionsExtensionsCatalog.ValueToKeyMappingEstimator%2A> transformation class.  The `MapValueToKey` returns an <xref:Microsoft.ML.Data.EstimatorChain%601> that will effectively be a pipeline. Name this `pipeline` as you will then append the trainer to it. Add the next line of code:
+
+[!code-csharp[MapValueToKey1](../../../samples/machine-learning/tutorials/TransferLearningTF/Program.cs#MapValueToKey1)]
+
+Your image processing pipeline uses pre-trained DNN featurizers for feature extraction. The images are loaded in memory as a Bitmap type. Next, you resize the images as the pre-trained model has a defined input image width and height. Finally, The `ImagePixelExtractingEstimator` extracts the pixels from the input images and converts them into a numeric vector. Add these image transforms as the next lines of code:
+
+[!code-csharp[ImageTransforms](../../../samples/machine-learning/tutorials/TransferLearningTF/Program.cs#ImageTransforms)]
+
+The `TensorFlowTransform` extracts specified outputs and scores a dataset using a pre-trained Tensorflow model. 
+
+[!code-csharp[ScoreTensorFlowModel](../../../samples/machine-learning/tutorials/TransferLearningTF/Program.cs#ScoreTensorFlowModel)]
+
+### Choose a learning algorithm
+
+To add the learning algorithm, call the `mlContext.MulticlassClassification.Trainers.LogisticRegression` wrapper method.  The `LogisticRegression` is appended to the `pipeline` and accepts the featurized `Title` and `Description` (`Features`) and the `Label` input parameters to learn from the historic data. You also need to map the label to the value to return to its original readable state. Do both of those actions with the following code:
```

Comment:
A few questions here should be answered. 
1. How and why are we trying to add these learners on top of an already pre-trained model? 
2. ML.NET supports a variety of learners for classification, how and why did we pick Logitistic Regression versus others
3. What is a pipeline? (follow concepts in the recommendation tutorial).